### ISSUE TITLE
Adiabatic surfaces are apparently not considered?

### ISSUE THREAD

**Issue by martin-mosteiro** [architecture-building-systems/CityEnergyAnalyst#3555]
**Describe the bug**
Per issue #443 and pull request #2416, adjacent buildings in CEA are _supposed to_ be considered as adiabatic surfaces. However, I tested simulating the demand for a building surrounded on three sides by other buildings compared to the same building after removing all surrounding buildings:
![Screenshot 2024-04-18 at 16 20 05](https://github.com/architecture-building-systems/CityEnergyAnalyst/assets/18548065/6cd34d47-eb24-4bb5-a90d-9c3ac6c96505) ![Screenshot 2024-04-18 at 16 20 30](https://github.com/architecture-building-systems/CityEnergyAnalyst/assets/18548065/536ff4d0-5374-4d69-857b-afe8232e99e4)

After running the solar radiation and the energy demand simulations for both cases, I get the following results:
![Screenshot 2024-04-18 at 16 18 45](https://github.com/architecture-building-systems/CityEnergyAnalyst/assets/18548065/30f89f27-2d4f-4179-9fe4-fb81056e96bc)
The plot on the top shows the results for the building surrounded on three sides, the bottom one shows the isolated one. As expected, the solar gains increase significantly when the building is isolated. However, the transmission losses through the envelope barely change – for example, the difference in transmission losses through the walls in January is 1.6% higher when the building is fully detached. 

Here's a scatter plot for the three most relevant parameters for both cases:
![image](https://github.com/architecture-building-systems/CityEnergyAnalyst/assets/18548065/e9675a38-24b3-4e99-a4a5-e09a6864d0f5)
The solar gains for the detached case go up by up to 8 times, but the transmission losses are basically linear.

Having investigated a bit more, the parameter implemented in PR #2416 appears to have been made optional for the sake of speed (`config.radiation.consider_intersections`) but as a result buildings are modeled as though they were fully surrounded by outdoor air.

**To Reproduce**
1. Create a scenario with adjacent buildings
2. Run the solar and demand simulations
3. Create a new scenario with the same buildings but remove attached buildings
4. Run the solar and demand simulations for the new scenario
5. Compare results

**Expected behaviour**
Surfaces attached to adjacent buildings should be treated as adiabatic.

**Comment by reyery** [comment]
Does having `config.radiation.consider_intersections` set to true solve this issue? I'm not too sure how much of a speed loss it has on runs.

**Comment by martin-mosteiro** [comment]
@reyery No, it doesn't. I believe the original fix did, but as implemented now I don't see any part of the code where adiabatic surfaces _would_ be considered in the simulation. I need to go through the entire thermal loads code to verify this, though (there are a number of `TODO` comments on the need to consider adiabatic surfaces, but I'm not sure if those refer to the entire calculation or specific parts of it). I'll keep posting here when/if I find out anything else.

**Comment by martin-mosteiro** [comment]
Actually... the results for the solar radiation script don't make any sense, either. The building in the example I showed above basically only has one exposed wall, facing west:
<img width="566" alt="Screenshot 2024-04-19 at 16 41 32" src="https://github.com/architecture-building-systems/CityEnergyAnalyst/assets/18548065/9ca3cb57-9651-4633-949e-abc9741953fb">
However, when running the radiation script I get:
- north: 1364.99 kWh/yr (15.8% of all radiation on wall/window surfaces)
- east: 630.4 kWh/yr (7.3% of all radiation on wall/window surfaces)
- south: 1630.92 kWh/yr (18.9% of all radiation on wall/window surfaces)
- west: 4992.48 kWh/yr (57.9% of all radiation on wall/window surfaces)
So... it looks like the results are **way** off. I could understand a small difference due to limitations in the script (say, a very small percentage of the solar irradiation on adjacent walls). But here more than 40% of the calculated irradiation falls on surfaces that shouldn't be getting any radiation at all?
I think `config.radiation.consider_intersections` _might_ solve this issue (I'll try it now), but if the error produced is so large I wonder if that parameter should be optional at all?

**Comment by martin-mosteiro** [comment]
`config.radiation.consider_intersections` indeed makes the results a lot more plausible:
- north: 0.0 kWh/yr
- east: 0.0 kWh/yr
- south: 682.51 kWh/yr (11.7%)
- west: 5148.63 kWh/yr (88.3%)
Interestingly, the irradiation on the west façade changed (maybe the bounces simulated in daysim changed after considering the intersections or something like that) but south-facing wall still receives a pretty big share of the irradiation.

**Comment by martin-mosteiro** [comment]
Hey @MatNif. Thanks for checking it out. The ledge should not be the issue, since there are no sensors on it. The sensors detecting the radiation are actually on the South-facing wall itself:
<img width="637" alt="Screenshot 2024-04-22 at 10 04 54" src="https://github.com/architecture-building-systems/CityEnergyAnalyst/assets/18548065/2fe3dda0-2dbc-4b7c-8451-596051ecfe06">

The good news is that considering overlaps (mostly) solves the issue of unrealistic solar irradiation and windows being placed in the wrong location (e.g., on the north- and east-facing walls in that building I showed), which is a step towards the right answer. The south-facing wall ends up getting an even larger wall area than the west facing one (which is further evidence that the radiation is not because of the little ledge on the top left of the picture), but I suppose that's an issue of bad input shapefiles, *not* CEA.

Next step would be to include adiabatic walls, which I believe is an easy fix (any wall with 0 irradiation should be adiabatic). I can implement that soon.

**Comment by reyery** [comment]
@martin-mosteiro there is a column named `intersection` in the radiation building geometry output (e.g. B1000_geometry.csv), I suppose that could be used to determine adiabatic surfaces.

**Comment by martin-mosteiro** [comment]
@reyery Thanks! That produces the same outcome but is a much more elegant (and consistent) solution :)

**Comment by martin-mosteiro** [comment]
Ok, so if the `consider_intersections` options is turned on, adjacent surfaces' areas are set to 0m2 in the outputs of the radiation script, so they're not taken into consideration in the demand script – i.e., they're already treated as adiabatic 👍 

So, to summarize: adjacent buildings are only considered adiabatic if the `consider_intersections` option is turned on. I think this is very counterintuitive. 👎

I would suggest the following:
1. rename the variable and/or its description to make it clear that that's the case; 
2. I would also suggest moving it to a more prominent location in the solar radiation tool, since it is not just some Daysim-specific input parameter and it affects not only the solar radiation results, but also the demand results; and
3. I would also consider making it True by default: sure, it can make simulations slower, but not considering attached walls is a bold assumption (see point 2).

I've made the change I specified in point 1. @reyery @MatNif please do let me know if you have any comments about points 2 and 3.

**Comment by MatNif** [comment]
Hey @martin-mosteiro ,

I absolutely agree with putting the setting in a more prominent location and renaming it. Setting it do True by default, I think we could discuss a bit more. As far as I understand, the demand script was originally developed to be a 1-zone building energy model. So I think that an argument can be made that, if we consider adjacent building footprints as two different buildings, we're effectively turning this 1-zone model into a multi-zone model (which might imply that we have other inaccuracies in modelling, not sure though since I don't know the demand script well enough).

Just food for thought. I think we can discuss this in more detail during the monthly meeting.

**Comment by reyery** [comment]
@martin-mosteiro I think I will set `consider_intersections` as default `True` for now, since I rather radiation generate reasonable results compared to fast but inaccurate. 

But @MatNif does bring up an interesting point for demand. However, I honestly do not know enough about demand to make any comments, so just let me know how you guys want to move forward with this.

**Comment by martin-mosteiro** [comment]
Sorry I missed your messages @reyery @MatNif. Thanks for your feedback.

Regarding the 1-zone model, the first issue is that if the adiabatic surfaces are not consistent with the standards, then they should be removed completely from the calculation, not be dependent to an optional parameter.
However, while the standards used in CEA (ISO 13790 and SIA 2044) include different considerations for the single- and multi-thermal zone cases, these differences do not relate to the thermal transmission coefficients to the outdoor environment, to the ground or surrounding buildings. More specifically:
- Section 6.2 in ISO 13790 defines the boundary of the building for the calculation as _"The boundary of the building for the calculation of the energy need for heating and/or cooling consists of all the building elements separating the conditioned space or spaces under consideration from the external environment (air, ground or water) *or from adjacent buildings* or unconditioned spaces."_
- Section 6.3 in ISO 13790 discusses single- vs. multi-zone modeling as follows: _"It may be necessary to partition a building into different zones, with separate calculation of the energy need for heating and cooling for each zone. [...]The boundary of a building zone consists of all the building elements separating the conditioned space or spaces under consideration from the external environment (air, ground or water), from adjacent conditioned zones, from adjacent buildings or from unconditioned spaces."_
So the definition of the building boundaries is independent of the definition of number of zones. The equations to calculate transmission heat transfer coefficients in section 8 therefore apply to any model, regardless of which of the options in Section 6.3 is selected. *So I would argue that making `consider_intersections` True by default is actually _more_ consistent with the standards than not doing so.*
Now _technically_ the standard does _not_ say the walls should be considered adiabatic, but a correction factor should be considered based on the temperature difference between the two buildings. So indeed, the current implementation in CEA is a simplification (note: the transmission to the ground in CEA is also simplified). I would argue that this is a common simplification in practice, though. But again, this is more a question about implementation of the standard than whether the intersections should be considered True by default or not.

My opinion is ultimately what Rey said: since not considering intersections could ultimately lead to incorrect results, I'd vote for making them `True` by default. There are however other options in CEA that are set to the quicker, less accurate option by default, though (e.g., the detailed ventilation calculation, the detailed thermal network model, etc.). So there's an argument to be made that default to `False` is okay. But I'll move it to a more prominent position so users can make their own decision on that tradeoff.

**Comment by martin-mosteiro** [comment]
TL;DR: per the standards, surfaces to adjacent conditioned or unconditioned buildings **should** be considered for the single thermal zone model, but it could be argued that making this default `False` is a CEA simplification for computational expense.

**Comment by martin-mosteiro** [comment]
I did some further testing to check whether the intersections indeed make the scripts much slower and... actually, they didn't (for me). Here's the cases I tested:
<img width="1130" alt="Screenshot 2024-05-01 at 13 46 08" src="https://github.com/architecture-building-systems/CityEnergyAnalyst/assets/18548065/a4ef877a-6695-4384-95a1-714931f8bab1">

So there are some considerable deviations, but definitely not a massive difference as the description of the variable hints at (up to 5x slower). Keep in mind that all Dutch scenarios have the same exact footprints, so any deviation in speed between the two "NL (18 bdgs.)" cases is likely to be because I was using my computer while testing (i.e., the deviation caused by me doing other stuff while the simulations run is comparable to the difference in simulation time for those small cases). Switzerland is reference-case-open, which has no possible overlaps, so again the difference in speed is likely not due to the script itself. 

It does seem to be the case that the bigger the case study, the bigger the difference in simulation time (since there are more intersections to process), though, so I will leave a message saying so in the variable description. But per the discussion above, I'll make the parameter `True` by default and make it more prominent so people can decide to toggle it off if they choose to.

**Comment by MatNif** [comment]
If the run-time is essentially the same, is it even worth keeping this as a user-setting? Is there any case where one would want to reduce the runtime of a simulation by 20% so badly, that people would tolerate such inaccuracies?

**Comment by reyery** [comment]
I have a feeling this parameter was introduced to make radiation faster before my work on making radiation use multiprocessing (based on the git blame timestamps). So I imagine it would probably have a greater effect on the time if it was running using a single process, and it does not really make sense now with multiple processes.

Whether or not to totally remove. I guess it could _maybe_ help with really large scenarios? Since it seems like the computational time increases exponentially without it. But then again this can be considered an edge case anyway, and users could just use the simplified model.

I guess if at the end of the day it does not conceptually makes sense, I say we remove it.

**Comment by reyery** [comment]
Solved in #3576
